src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Nullable.cs

```diff
@@ -0,0 +1,907 @@
+        {
+            var source =
+@"delegate void D<T>(T t);
+class C
+{
+    static void F<T>(D<T> d)
+    {
+    }
+    static void G()
+    {
+        F((object? o) => { });
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
+            // PROTOTYPE(NullableReferenceTypes): Use AssertNullableAttribute(method.Parameters[0].GetAttributes()). 
+            AssertAttribute(comp);
+        }
+
+        [Fact]
+        public void EmitAttribute_LocalFunctionReturnType()
+        {
+            var source =
+@"class C
+{
+    static void M()
+    {
+        object?[] L() => throw new System.NotImplementedException();
+        L();
+    }
+}";
+            CompileAndVerify(
+                source,
+                parseOptions: TestOptions.Regular8,
+                options: TestOptions.DebugDll.WithMetadataImportOptions(MetadataImportOptions.All),
+                symbolValidator: module =>
+                {
+                    var method = module.ContainingAssembly.GetTypeByMetadataName("C").GetMethod("<M>g__L|0_0");
+                    AssertNullableAttribute(method.GetReturnTypeAttributes());
+                });
+        }
+
+        [Fact]
+        public void EmitAttribute_LocalFunctionParameters()
+        {
+            var source =
+@"class C
+{
+    static void M()
+    {
+        void L(object? x, object y) { }
+        L(null, 2);
+    }
+}";
+            CompileAndVerify(
+                source,
+                parseOptions: TestOptions.Regular8,
+                options: TestOptions.DebugDll.WithMetadataImportOptions(MetadataImportOptions.All),
+                symbolValidator: module =>
+                {
+                    var method = module.ContainingAssembly.GetTypeByMetadataName("C").GetMethod("<M>g__L|0_0");
+                    AssertNullableAttribute(method.Parameters[0].GetAttributes());
+                    AssertNoNullableAttribute(method.Parameters[1].GetAttributes());
+                });
+        }
+
+        [Fact]
+        public void EmitAttribute_ExplicitImplementationForwardingMethod()
+        {
+            var source0 =
+@"public class A
+{
+    public object? F() => null;
+}";
+            var comp0 = CreateStandardCompilation(source0, parseOptions: TestOptions.Regular8);
+            var ref0 = comp0.EmitToImageReference();
+            var source =
+@"interface I
+{
+    object? F();
+}
+class B : A, I
+{
+}";
+            CompileAndVerify(
+                source,
+                additionalRefs: new[] { ref0 },
+                parseOptions: TestOptions.Regular8,
+                options: TestOptions.DebugDll.WithMetadataImportOptions(MetadataImportOptions.All),
+                symbolValidator: module =>
+                {
+                    var method = module.ContainingAssembly.GetTypeByMetadataName("B").GetMethod("I.F");
+                    AssertNullableAttribute(method.GetReturnTypeAttributes());
```

Comment:
>AssertNullableAttribute(method.GetReturnTypeAttributes()); [](start = 20, length = 58)

Please assert that we don't have the attribute on the method itself, same for other test scenarios.

Reply:
Added.